auth_ldaps/models/res_company_ldap.py

```diff
@@ -0,0 +1,42 @@
+# Copyright (C) Creu Blanca
+# Copyright (C) 2018 Brainbean Apps
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+
+import logging
+
+import ldap
+
+from odoo import fields, models
+
+_logger = logging.getLogger(__name__)
+
+
+class CompanyLDAP(models.Model):
+    _inherit = "res.company.ldap"
+    _description = "Company LDAP configuration"
+
+    is_ssl = fields.Boolean(string="Use LDAPS", default=False)
+    skip_cert_validation = fields.Boolean(
+        string="Skip certificate validation", default=False
+    )
+
+    def _get_ldap_dicts(self):
+        res = super()._get_ldap_dicts()
+        for rec in res:
+            ldap = self.sudo().browse(rec["id"])
+            rec["is_ssl"] = ldap.is_ssl or False
+            rec["skip_cert_validation"] = ldap.skip_cert_validation or False
+        return res
+
+    def _connect(self, conf):
+        if conf["is_ssl"]:
+            uri = "ldaps://%s:%d" % (conf["ldap_server"], conf["ldap_server_port"])
+            connection = ldap.initialize(uri)
+            if conf["skip_cert_validation"]:
+                connection.set_option(ldap.OPT_X_TLS_REQUIRE_CERT, ldap.OPT_X_TLS_ALLOW)
+            connection.set_option(ldap.OPT_X_TLS_NEWCTX, 0)
```

Comment:
It was the same on version 13.0 So I won't block on this one, I'm wondering why setting `ldap.OPT_X_TLS_NEWCTX` option if there are no changes in the fresh initialized connection ?

As far I understand [documentation](https://www.python-ldap.org/en/python-ldap-3.3.0/reference/ldap.html?highlight=set_option#tls-options) this is required while changing TLS context, so I would move this setting likes this:
```suggestion
            if conf["skip_cert_validation"]:
                connection.set_option(ldap.OPT_X_TLS_REQUIRE_CERT, ldap.OPT_X_TLS_ALLOW)
                connection.set_option(ldap.OPT_X_TLS_NEWCTX, 0)
```